jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -75,9 +79,36 @@
       throws GeneralSecurityException {
     // Do not use {@link NetHttpTransport}. See {@link getConnectionFactory} for details.
     HttpTransport transport = new ApacheHttpTransport.Builder().doNotValidateCertificate().build();
+    addProxyCredentials(transport);
     return url -> new Connection(url, transport);
   }
 
+  private static void addProxyCredentials(HttpTransport transport) {
+    DefaultHttpClient httpClient =
+        (DefaultHttpClient) ((ApacheHttpTransport) transport).getHttpClient();
+
+    if (System.getProperty("http.proxyUser") != null) {
+      httpClient
+          .getCredentialsProvider()
+          .setCredentials(
+              new AuthScope(
+                  System.getProperty("http.proxyHost"),
+                  Integer.parseInt(System.getProperty("http.proxyPort"))),
+              new UsernamePasswordCredentials(
+                  System.getProperty("http.proxyUser"), System.getProperty("http.proxyPassword")));
+    } else if (System.getProperty("https.proxyUser") != null) {
```

Comment:
Originally I thought to have separate sets of properties for http and https. But I just learned that Maven active proxy applies to both http and https, so we need only a single set of credential properties. I will change the code to only use http.proxyUser and http.proxyPassword, and apply them on AuthScope.ANY.

Reply:
Sorry if I'm not getting this correct:

What you suggested seems to make sense when they were given through the Maven settings file. What if both `http.proxyUser` and `https.proxyUser` are defined via command line system properties instead of the Maven settings file? (A Gradle project is an example, but it's also a possible scenario in Maven.)

Reply:
And isn't it that setting the scope with the given host is the correct way? I wonder if `AuthScope.ANY` may interfere with Jib setting auth for registries.

Related, what's the end result of calling `.setCredentials()`? For example, does it end up with setting the `Proxy-Authorization: Basic xxx`? (Or, the `Authorization:` header?)

Reply:
Actually, `mavenSettings` may contain multiple proxy settings, and only the last HTTP and HTTPS settings win with this code. For this to work properly, I think we need to call `.setCredentials()` for each proxy settings in `mavenSettings`. Looks like we need to pass down the Maven proxy auth info to the `Connection` class, which will be a bit involved.

Reply:
Ok, indeed if organization has proxy server(s) with different credential pairs for accessing proxy over https, and accessing over http, then we would need to have 2 separate credential pairs. I guess in most cases - like my organization - it would be the same credential pair. So would be nice if you could also specify just a single credential pair indifferent of the protocol. I mean, by default use http.proxyUser for both protocols, unless https.proxyUser is explicitly specified. This would also align with sending just single credential pair (http.proxyUser) from active proxy in Maven settings.

Reply:
Suggestion:

```suggestion
  private static void addProxyCredentials(HttpTransport transport) {
    DefaultHttpClient httpClient =
        (DefaultHttpClient) ((ApacheHttpTransport) transport).getHttpClient();

    boolean httpProxy = System.getProperty("http.proxyHost") != null;
    boolean httpCreds = System.getProperty("http.proxyUser") != null;
    boolean httpsProxy = System.getProperty("https.proxyHost") != null;
    boolean httpsCreds = System.getProperty("https.proxyUser") != null;
    if (httpProxy && httpCreds) {
      httpClient
              .getCredentialsProvider()
              .setCredentials(
                      new AuthScope(System.getProperty("http.proxyHost"), Integer.parseInt(System.getProperty("http.proxyPort"))),
                      new UsernamePasswordCredentials(
                              System.getProperty("http.proxyUser"), System.getProperty("http.proxyPassword")));
    }
    if (httpsProxy && (httpsCreds || httpCreds)) {
      httpClient
              .getCredentialsProvider()
              .setCredentials(
                      new AuthScope(System.getProperty("https.proxyHost"), Integer.parseInt(System.getProperty("https.proxyPort"))),
                      new UsernamePasswordCredentials(
                              httpsCreds ? System.getProperty("https.proxyUser") : System.getProperty("http.proxyUser"), 
                              httpsCreds ? System.getProperty("https.proxyPassword") : System.getProperty("http.proxyPassword")));
    }
  }
```

Reply:
Never mind my last remark on Maven active proxy. DefaultProxySelector currently relied on, selects proxy protocol properties matching request protocol. So indeed its better to take into account multiple active proxies for both http and https requests.

Reply:
Please have a look on my last commit. It takes into account multiple Maven active proxies and registers proxy credentials per protocol.